Add check list editor tests

ievacerny opened the pull request:
Adds tests for check list editor

ievacerny commented on traitsui/tests/editors/test_check_list_editor.py:
```diff
@@ -0,0 +1,512 @@
+                style=style,
+            )
+        )
+        model = ListModel(possible_values=["one", "two"])
```
That was left in by mistake, removed now

ievacerny commented on traitsui/tests/editors/test_check_list_editor.py:
```diff
@@ -0,0 +1,512 @@
+        return combobox.currentText()
+
+
+def set_combobox_index(editor, idx):
```
For wx the function is essentially the same as in `test_enum_editor` so I would like to keep the name the same. The difference is that I can't test setting the index for qt and have to pass in the editor rather than combobox control to do an equivalent test. I update the docstring, hopefully it explains it better.

kitchoi replied:
Yep. It helps. Thank you!

ievacerny commented:
The behaviour of combobox (simple check list editor) when the list is empty is not well defined, so I modified the tests to have a non empty default list.

Also added a temporary fix to avoid #851 so that other tests (enum and csv editor) could run properly.

kitchoi commented:
> The behaviour of combobox (simple check list editor) when the list is empty is not well defined, so I modified the tests to have a non empty default list.

Ah I hadn't noticed that. Could you explain a bit more please? What causes the behaviour to be ill defined?

ievacerny commented:
> > The behaviour of combobox (simple check list editor) when the list is empty is not well defined, so I modified the tests to have a non empty default list.
> 
> Ah I hadn't noticed that. Could you explain a bit more please? What causes the behaviour to be ill defined?

Simple check list editor acts more like an enum editor than a list editor - only one valid value can be selected. Empty list is not a valid value so `update_editor` fails to set a selection and the value shown depends on the toolkit and platform implementation. Usually the first item on the list of valid values is shown, except for windows wx implementation as shown by test failures.

kitchoi commented:
I see. Thank you.
I think it is possible to call `QtComboBox.setCurrentText(-1)` and `wx.Choice.SetSelection(wx.NOT_FOUND)` for that scenario. I think that should be a separate issue.
